### PR TITLE
Minimal example of gen-ai-rago

### PR DESCRIPTION
**Pull request by dumpsys**
The code is taken from https://github.com/GDP-ADMIN/gen-ai-internal/tree/main/examples/gen-ai-rago
with dependency to binary sdk

### CODE REVIEW

File: examples/gen-ai-hello-world/local-start.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+main() {
+    # Main function to orchestrate the deployment script.
+    log "Checking gen-ai-hello-world example requirements..."
+    check_requirements
+    deactivate_conda
+
+    log "Running gen-ai-hello-world example..."
+    poetry config http-basic.gen-ai-internal oauth2accesstoken $(gcloud auth print-access-token)
```
Comment (timotiusnc) [inline examples/gen-ai-hello-world/local-start.sh:102]: How about we use
```
export POETRY_HTTP_BASIC_GEN_AI_INTERNAL_USERNAME=oauth2accesstoken
export POETRY_HTTP_BASIC_GEN_AI_INTERNAL_PASSWORD=$(gcloud auth print-access-token)
```
like in https://github.com/GDP-ADMIN/gen-ai-internal/blob/main/libs/BINARY.md#installation-using-poetry

So we also give example to developers on how to integrate it with their CI system.

File: examples/gen-ai-hello-world/local-start.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        fi
+        if [[ "$(printf '%s\n' "$required_version" "$current_version" | sort -V | head -n1)" != "$required_version" ]]; then
+            handle_error "$cmd version $required_version or above is required. You have version $current_version."
+        fi
+    fi
+
+    log "$cmd is installed and meets the required version ($required_version). Current version: $current_version."
+}
+
+check_requirements() {
```
Comment (timotiusnc) [inline examples/gen-ai-hello-world/local-start.sh:56]: Let's also check for `gcloud` pre-requisite.

Comment (dumpsys) [reply examples/gen-ai-hello-world/local-start.sh:57]: updated ya [5358986](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/5358986f97b9fee950b140d31788e934846b9137)
I also slightly modify the regex to grep the version

File: examples/gen-ai-hello-world/local-start.sh (hunk 1 above)
Comment (dumpsys) [reply examples/gen-ai-hello-world/local-start.sh]: updated [6ed2fdc](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/6ed2fdc188b4c17f33edbd4223b60b6f9f632bf4)

File: examples/gen-ai-hello-world/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+> [!NOTE]
+> The prerequisites are checked automatically when you run the script.
+
+## Running the code
+
+1. Configure Environment: Copy `.env.example` to `.env` and set up the environment variables.
+2. Execute the script:
+
+```
```
Comment (felicia-limanta-gdp) [inline examples/gen-ai-hello-world/README.md:29]:
```suggestion
```bash
```

File: examples/gen-ai-hello-world/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+local-start.sh
```
Comment (felicia-limanta-gdp) [inline examples/gen-ai-hello-world/README.md:30]: krn mau execute 

```suggestion
./local-start.sh
```

sama, local-start-nya perlu dikasih chmod +x dulu ngga ya?

File: examples/gen-ai-hello-world/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+   - **Operating System**: Linux
+   - **Architecture**: x86_64
+   - **Python Version**: 3.11
+
+> [!NOTE]
+> The prerequisites are checked automatically when you run the script.
+
+## Running the code
+
+1. Configure Environment: Copy `.env.example` to `.env` and set up the environment variables.
```
Comment (felicia-limanta-gdp) [inline examples/gen-ai-hello-world/README.md:26]: Kayanya perlu ditulis navigate ke `gen-ai-hello-world` dulu.
Sama mungkin bias ditambahin cara copy .env.example ke .env pakai script (?)

File: examples/gen-ai-hello-world/.env.example [hunk 6]
```diff
@@ -0,0 +1,2 @@
+OPENAI_API_KEY = "<YOUR_OPENAI_API_KEY>"
```
Comment (felicia-limanta-gdp) [inline examples/gen-ai-hello-world/.env.example:1]: Ini di antara valuenya perlu petik kah?

```suggestion
OPENAI_API_KEY = <YOUR_OPENAI_API_KEY>
```

File: examples/gen-ai-hello-world/.env.example [hunk 7]
```diff
@@ -0,0 +1,2 @@
...
+LANGUAGE_MODEL = "<VALID_OPENAI_LANGUAGE_MODEL_NAME>"
```
Comment (felicia-limanta-gdp) [inline examples/gen-ai-hello-world/.env.example:2]: Ini minor banget, tpi mungkin bisa dikasih komen penjelasan gini 
```suggestion
LANGUAGE_MODEL = "<VALID_OPENAI_LANGUAGE_MODEL_NAME>" # for example gpt-4, gpt-4-turbo, ...
```

File: examples/gen-ai-hello-world/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
+# GenAI RAGO Example
+
+This is an example of how to use the gllm-pipeline library to build a RAG pipeline.
+
+## Prerequisites
+
+1. Python v3.11 or above.
```
Comment (felicia-limanta-gdp) [inline examples/gen-ai-hello-world/README.md:7]: Instalasi python mau diarahin pakai miniconda kah?

File: examples/gen-ai-hello-world/README.md (hunk 5 above)
Comment (dumpsys) [reply examples/gen-ai-hello-world/README.md:28]: updated ya [19ef938](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/19ef938941dbc796abdd7c2184a9f28c5652aef3)

File: examples/gen-ai-hello-world/.env.example (hunk 7 above)
Comment (dumpsys) [reply examples/gen-ai-hello-world/.env.example]: ini sudah di sini [19ef938](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/19ef938941dbc796abdd7c2184a9f28c5652aef3)

File: examples/gen-ai-hello-world/README.md (hunk 8 above)
Comment (dumpsys) [reply examples/gen-ai-hello-world/README.md:7]: updated [19ef938](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/19ef938941dbc796abdd7c2184a9f28c5652aef3)

File: examples/gen-ai-hello-world/.env.example (hunk 6 above)
Comment (dumpsys) [reply examples/gen-ai-hello-world/.env.example:1]: ah sorry, yang ini terlewat, aku update lagi

Comment (dumpsys) [reply examples/gen-ai-hello-world/.env.example]: tanda petik di remove di sini [aefbb73](https://github.com/GDP-ADMIN/gen-ai-examples/pull/1/commits/aefbb737387f38ad690c35babf94a4a03695f1e4)

File: examples/gen-ai-hello-world/README.md (hunk 4 above)
Comment (dumpsys) [reply examples/gen-ai-hello-world/README.md]: oh iya, ini file mode-nya sudah executable ya. kalau kucek seperti ini `-rwxr-xr-x@`, jadi mestinya bisa langsung `./local-start.sh`

Comment (felicia-limanta-gdp) [reply examples/gen-ai-hello-world/README.md]: oh okeii

**Pull request merged**